pkg/api/groups.go

```diff
@@ -57,39 +75,77 @@ func (a *API) GetGroup(w http.ResponseWriter, r *http.Request, groupId string) {
 
 	_, err := a.DB.Query(ctx, &q)
 	// return a 404 if the user was not found.
-	if errors.As(err, &identity.UserNotFoundError{}) {
+	if err == ddb.ErrNoItems {
 		err = apio.NewRequestError(err, http.StatusNotFound)
 	}
 
 	if err != nil {
 		apio.Error(ctx, w, err)
 		return
 	}
-
 	apio.JSON(ctx, w, q.Result.ToAPI(), http.StatusOK)
 }
 
 // Create Group
 // (POST /api/v1/admin/groups)
+// Creates an internal group not connected to any identiy provider in dynamodb
 func (a *API) CreateGroup(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
-	if a.Cognito == nil {
-		apio.ErrorString(ctx, w, "api not available", http.StatusBadRequest)
-		return
-	}
 	var createGroupRequest types.CreateGroupJSONRequestBody
 	err := apio.DecodeJSONBody(w, r, &createGroupRequest)
 	if err != nil {
 		apio.Error(ctx, w, apio.NewRequestError(err, http.StatusBadRequest))
 		return
 	}
-	group, err := a.Cognito.CreateGroup(ctx, cognitosvc.CreateGroupOpts{
-		Name:        createGroupRequest.Name,
-		Description: aws.ToString(createGroupRequest.Description),
-	})
+
+	group := identity.Group{}
+	if createGroupRequest.Id != nil {
+		//update existing group
+		group = identity.Group{
+			ID:          *createGroupRequest.Id,
+			IdpID:       createGroupRequest.Name,
+			Name:        createGroupRequest.Name,
+			Description: aws.ToString(createGroupRequest.Description),
+			Status:      types.IdpStatusACTIVE,
+			Source:      identity.INTERNAL,
+			Users:       createGroupRequest.Members,
+			CreatedAt:   time.Now(),
```

Comment:
created at should not change on an update

I would also prefer to see these as a put and post operate seperately